pkg/cmd/reset.go

```diff
@@ -85,6 +88,21 @@ func resetCmd(rootFlags *pflag.FlagSet) *cobra.Command {
 
 // runReset resets all machines provisioned by KubeOne
 func runReset(opts *resetOpts) error {
+	// warn of explicit confirmation flag starting with KubeOne 1.3
+	// We need to use semver.Compare here, because semver.Validate
+	// via constraints do not support pre-release versions
+	c, err := semver.NewVersion("1.3")
+	if err != nil {
+		return err
+	}
+	v, err := semver.NewVersion(version)
+	if err != nil {
+		return err
+	}
+	if result := c.Compare(v); result > 0 {
+		fmt.Println("WARNING: This command will require an explicit confirmation starting with KubeOne version 1.3")
```

Comment:
I have two comments regarding this line:
1) I think that we should use the logger instead of `fmt.Println`. You should be able to use the logger such as `s.Logger.Warnln` after running `opts.BuildState`
2) Let's change the message to make it clear that 1.3 is the next minor version
```suggestion
		fmt.Println("This command will require an explicit confirmation starting with the next minor release (v1.3)")
```

Reply:
@xmudrii  I am a little bit confused now, because @kron4eg  says I should remove the version check entirely and instead add an auto-approve NOOP flag to the reset command.

Can one of you tell me what I shall do now? :D

Reply:
I mean, the message about 1.3 should be in place, it's just the check that should be removed. Plus a NOOP flag (to show it off in the --help)

Reply:
@shibumi We should do both. There should be a no-op flag, but also a warning that we would show unconditionally (without version check). The warning should say that this command will require an explicit confirmation and that we added a no-op flag that will be used as of the next version to skip the confirmation.

Reply:
@kron4eg  @xmudrii  ah ok, now I got it. Sorry for the confusion folks :)